### PR TITLE
SPEC-1044: Clarify that retryable writes may fail with FCV 3.4 shard

### PR DESCRIPTION
**Pull request by jmikola**
cc: @renctan

### CODE REVIEW

File: source/retryable-writes/retryable-writes.rst [hunk 1]
```diff
@@ -10,7 +10,7 @@ Retryable Writes
 :Status: Draft (awaiting reference implementation)
```
Comment (ajdavis) [inline source/retryable-writes/retryable-writes.rst]: Let's update the Status next.

File: source/retryable-writes/retryable-writes.rst [hunk 2]
```diff
@@ -126,13 +126,13 @@ writes were not enabled. Drivers MUST NOT include a transaction ID in the write
 command and MUST not retry the command under any circumstances.
 
 In a shard cluster, it is possible that mongos may appear to support retryable
```
Comment (ajdavis) [inline source/retryable-writes/retryable-writes.rst]: "sharded cluster"

**Comment by renctan** [comment]
lgtm

File: source/retryable-writes/retryable-writes.rst [hunk 3]
```diff
@@ -126,13 +126,13 @@ writes were not enabled. Drivers MUST NOT include a transaction ID in the write
...
-writes but an individual shard in the cluster does not (e.g. replica set's
-feature compatibility version is downgraded, standalone is added a new shard).
-In these rare cases, a write command that fans out to a shard that does not
-support retryable writes may partially fail and an error may be reported in the
-write result from mongos (e.g. ``writeErrors`` array in the bulk write result).
-This does not constitute a retryable error. Drivers MUST relay such errors to
-the user.
+writes but one or more shards in the cluster does not (e.g. replica set shard
```
Comment (ajdavis) [inline source/retryable-writes/retryable-writes.rst]: "one or more shards in the cluster do not"

**Pull request merged**